From 1cd3c6cba13a5d6e234396bd35b55f6e250cff18 Mon Sep 17 00:00:00 2001
From: "Andrea Grazioso (agr-odoo)" <agr@odoo.com>
Date: Thu, 1 Aug 2019 11:54:32 +0000
Subject: [PATCH] [FIX] account_reports: Auto follow for statement sender

From Accounting / Sales / Customer Statement, selecting a
statement in need of action and clicking on "send email" do send the
message correctly, as it get logged in the chatter.

Before the commit: The user who clicked on "send email" is not
automatically added as follower of the thread, which could be
inpractical, because reply from the customer may go unnoticed.

After: User who click on "send email" is added as follower of the
discussion. This is an expected behavior in case of email sending,
because it helps following the partner reply.

OPW-2041448

closes odoo/enterprise#4996

Signed-off-by: Lucas Perais (lpe) <lpe@odoo.com>
---
 account_reports/models/account_followup_report.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/account_reports/models/account_followup_report.py b/account_reports/models/account_followup_report.py
index 000c183d74..c922de6b74 100644
--- a/account_reports/models/account_followup_report.py
+++ b/account_reports/models/account_followup_report.py
@@ -360,7 +360,7 @@ class account_report_context_followup(models.TransientModel):
             body_html = self.with_context(public=True, mode='print').get_html()
             msg = self._get_email_sent_log()
             msg += '<br>' + ustr(body_html)
-            msg_id = self.partner_id.message_post(body=msg, subtype='account_reports.followup_logged_action')
+            msg_id = self.partner_id.message_post(body=msg, subtype='account_reports.followup_logged_action', message_type='email')
             email = self.env['mail.mail'].create({
                 'mail_message_id': msg_id.id,
                 'subject': _('%s Payment Reminder') % (self.env.user.company_id.name) + ' - ' + self.partner_id.name,
-- 
2.17.1

